### PR TITLE
DAG state channel infrastructure and startup flow

### PR DESCRIPTION
**Pull request by TheMMaciek**

### CODE REVIEW

File: modules/dag-l1/src/main/scala/org/tessellation/dag/l1/Main.scala [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+  def run(method: Run, sdk: SDK[IO]): Resource[IO, Unit] = {
+    import sdk._
+
+    val cfg = method.appConfig
+
+    Database.forAsync[IO](cfg.dbConfig).flatMap { implicit database =>
+      for {
+        _ <- IO.unit.asResource
+        configs = Configs.make
```
Comment (marcinwadon) [inline modules/dag-l1/src/main/scala/org/tessellation/dag/l1/Main.scala:35]: shouldn't config came from the cli method? take a look how it's done in `core`

File: modules/dag-l1/src/main/scala/org/tessellation/dag/l1/Main.scala [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+          .start(storages, services, queues, p2pClient, rumorHandler, nodeId, cfg)
+          .asResource
+
+        api = HttpApi.make[IO](storages, queues, keyPair.getPrivate, services, sdkPrograms)
+        _ <- MkHttpServer[IO].newEmber(ServerName("public"), cfg.httpConfig.publicHttp, api.publicApp)
+        _ <- MkHttpServer[IO].newEmber(ServerName("p2p"), cfg.httpConfig.p2pHttp, api.p2pApp)
+        _ <- MkHttpServer[IO].newEmber(ServerName("cli"), cfg.httpConfig.cliHttp, api.cliApp)
+        stateChannel = new StateChannel(configs, keyPair, p2pClient, queues, nodeId, services, storages, validators)
+        _ <- {
+          {
```
Comment (marcinwadon) [inline modules/dag-l1/src/main/scala/org/tessellation/dag/l1/Main.scala:55]: why double brackets?

File: modules/dag-l1/src/main/scala/org/tessellation/dag/l1/StateChannel.scala [hunk 3]
```diff
@@ -13,73 +13,60 @@ import cats.syntax.traverse._
...
+      selfId,
+      storages.transaction
     )
 
   private val ownerBlockConsensusInputs: Stream[F, OwnerBlockConsensusInput] = Stream
     .awakeEvery(5.seconds)
-    .evalFilter(_ => canStartOwnConsensus(consensusStorage, nodeStorage, clusterStorage, consensusConfig.peersCount))
+    .evalFilter(
+      _ => canStartOwnConsensus(storages.consensus, storages.node, storages.cluster, configs.consensus.peersCount)
+    )
```
Comment (marcinwadon) [inline modules/dag-l1/src/main/scala/org/tessellation/dag/l1/StateChannel.scala:65]: if so, I'd use curly brackets:
```scala
.evalFilter { _ =>
  canStart...
}
```

File: modules/dag-l1/src/main/scala/org/tessellation/dag/l1/modules/Configs.scala [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+object Configs {
+  def make: Configs = new Configs(BlockValidatorConfig(), ConsensusConfig(), TipsConfig()) {}
+}
+
+sealed abstract class Configs private (
+  val blockValidator: BlockValidatorConfig,
+  val consensus: ConsensusConfig,
+  val tips: TipsConfig
+)
```
Comment (marcinwadon) [inline modules/dag-l1/src/main/scala/org/tessellation/dag/l1/modules/Configs.scala:15]: I'd merge that into app config to have it in one place (see my other comment)

File: modules/dag-l1/src/main/scala/org/tessellation/dag/l1/Main.scala (hunk 2 above)
Comment (TheMMaciek) [reply modules/dag-l1/src/main/scala/org/tessellation/dag/l1/Main.scala]: Leftover, not needed.

**Review by marcinwadon — APPROVED** [review]

**Pull request merged**